freeipa/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -35,17 +35,24 @@
 
     private JsonRpcHttpClient jsonRpcHttpClient;
 
+    private String user;
```

Comment:
You are correct, this is the admin user. In this case the admin user can do anything but they have the grant rights to themself first. By default the admin user does not have access rights to retrieve existing keytabs for service principals without invalidating prior keytabs. The admin username is tracked so that it can grant itself access to perform the keytab retrieval operation.

Do we really need this here? I wanted to avoid having more than a single location that is hardcoded as "admin".

Reply:
https://github.com/hortonworks/cloudbreak/blob/master/freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/FreeIpaClientFactory.java#L59
I would extract the "admin" to a public static final String field and pass as parameter to the necessary parameter. This user is used only that 1 time.

Reply:
Since the `FreeIpaClientFactory` is injected, I did something similar to what you recommended. I added a `private static final String` with an accessor.